tests/Unit/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/Test_GaugeWave.cpp

```diff
@@ -0,0 +1,296 @@
+
+  // Check quantities derivable from spatial metric, lapse, shift, and their
+  // derivatives
+  const auto& inv_gamma =
+      get<gr::Tags::InverseSpatialMetric<3, Frame::Inertial, DataType>>(vars);
+  const auto& sqrt_det_gamma =
+      get<gr::Tags::SqrtDetSpatialMetric<DataType>>(vars);
+  const auto& det_and_inverse = determinant_and_inverse(gamma);
+  const auto& expected_inv_gamma = det_and_inverse.second;
+  const auto& expected_sqrt_det_gamma = sqrt(get(det_and_inverse.first));
+  CHECK_ITERABLE_APPROX(inv_gamma, expected_inv_gamma);
+  CHECK_ITERABLE_APPROX(get(sqrt_det_gamma), expected_sqrt_det_gamma);
+
+  const auto& K =
+      get<gr::Tags::ExtrinsicCurvature<3, Frame::Inertial, DataType>>(vars);
+  const auto& expected_K =
+      gr::extrinsic_curvature(lapse, shift, d_shift, gamma, dt_gamma, d_gamma);
+  CHECK_ITERABLE_APPROX(K, expected_K);
+
+  // Check that we can retrieve tags individually from the solution
+  tmpl::for_each<gr::Solutions::GaugeWave::tags<DataType>>(
+      [&solution, &vars, &x, &t ](auto tag_v) noexcept {
+        using Tag = tmpl::type_from<decltype(tag_v)>;
+        CHECK_ITERABLE_APPROX(get<Tag>(vars), get<Tag>(solution.variables(
+                                                  x, t, tmpl::list<Tag>{})));
+      });
+
+  // Check with random values
+  pypp::SetupLocalPythonEnvironment local_python_env{
+      "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/"};
+  pypp::check_with_random_values<1, GaugeWaveProxy::variables_tags<DataType>>(
+      &GaugeWaveProxy::test_variables<DataType>,
+      GaugeWaveProxy(amplitude, wavelength), "TestFunctions",
+      {"gauge_wave_lapse", "gauge_wave_dt_lapse", "gauge_wave_d_lapse",
+       "gauge_wave_shift", "gauge_wave_dt_shift", "gauge_wave_d_shift",
+       "gauge_wave_spatial_metric", "gauge_wave_dt_spatial_metric",
+       "gauge_wave_d_spatial_metric", "gauge_wave_sqrt_det_spatial_metric",
+       "gauge_wave_extrinsic_curvature", "gauge_wave_inverse_spatial_metric"},
+      {{{1.0, 20.0}}}, std::make_tuple(amplitude, wavelength), used_for_size);
+}
+
+void test_serialize() noexcept {
+  const gr::Solutions::GaugeWave solution(3.0, 4.0);
+  test_serialization(solution);
+}
+
+void test_copy_and_move() noexcept {
+  gr::Solutions::GaugeWave solution(3.0, 4.0);
+  test_copy_semantics(solution);
+  auto solution_copy = solution;
+  // clang-tidy: std::move of trivially copyable type
+  test_move_semantics(std::move(solution), solution_copy);  // NOLINT
+}
+
+void test_construct_from_options() {
+  Options<tmpl::list<GaugeWaveOptionTag>> opts("");
+  opts.parse(
+      "GaugeWaveOptionTag:\n"
+      "  Amplitude: 0.25\n"
+      "  Wavelength: 4.0");
+  CHECK(opts.get<GaugeWaveOptionTag>() == gr::Solutions::GaugeWave(0.25, 4.0));
+
+  Options<tmpl::list<GaugeWaveOptionTag>> opts_default("");
+  opts_default.parse("GaugeWaveOptionTag:\n");
+  CHECK(opts_default.get<GaugeWaveOptionTag>() ==
+        gr::Solutions::GaugeWave(1.0, 1.0));
```

Comment:
And do a `test_gauge_wave(opts.get<GaugeWaveOptionTag>(), DataVector{5});`

I think the default check here is fine as is :)